### PR TITLE
Fix lightning-persister tests for upcoming rustc changes

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Upcoming versions of rustc will break existing code (:scream:), so we need to tweak two tests in `lightning-persister` to work around this. This should fix tests on rustc-beta.

### CODE REVIEW

File: lightning-persister/src/util.rs [hunk 1]
```diff
@@ -135,7 +135,7 @@ mod tests {
 		// Create the channel data file and make it a directory.
 		fs::create_dir_all(get_full_filepath(path.clone(), filename.to_string())).unwrap();
 		match write_to_file(path.clone(), filename.to_string(), &test_writeable) {
-			Err(e) => assert_eq!(e.kind(), io::ErrorKind::Other),
+			Err(e) => assert_eq!(e.raw_os_error(), Some(libc::EISDIR)),
```
Comment (jkczyz) [inline lightning-persister/src/util.rs:138]: Consider adding a `TODO` to switch back to `kind` once possible. Though I guess that is a long way out. 😕

**Review by jkczyz — COMMENTED** [review]
Looks like `http::client_tests::connect_to_unresolvable_host` has some platform-specific errors. 😞

File: lightning-persister/src/util.rs (hunk 1 above)
Comment (TheBlueMatt) [reply lightning-persister/src/util.rs:138]: Yea, that's gonna require at least rustc-now, so I guess in at least two years per the rust-bitcoin MSRV policy, assuming distros upgrade. I think this is fine, in any case, at least it won't break.

File: lightning-block-sync/src/http.rs [hunk 2]
```diff
@@ -636,7 +636,7 @@ pub(crate) mod client_tests {
 	#[test]
 	fn connect_to_unresolvable_host() {
 		match HttpClient::connect(("example.invalid", 80)) {
-			Err(e) => assert_eq!(e.kind(), std::io::ErrorKind::Other),
+			Err(_) => { /* rust libstd currently doesn't have a specific error for Host Not Found for us to match against */ },
```
Comment (valentinewallace) [inline lightning-block-sync/src/http.rs:639]: Thoughts on this patch or similar:

```
diff --git a/lightning-block-sync/src/http.rs b/lightning-block-sync/src/http.rs
index 1f30bea1..93f7bafb 100644
--- a/lightning-block-sync/src/http.rs
+++ b/lightning-block-sync/src/http.rs
@@ -636,7 +636,9 @@ pub(crate) mod client_tests {
        #[test]
        fn connect_to_unresolvable_host() {
                match HttpClient::connect(("example.invalid", 80)) {
-                       Err(_) => { /* rust libstd currently doesn't have a specific error for Host Not Found for us to match against */ },
+                       Err(e) => {
+                               assert_eq!(e.to_string(), "failed to lookup address information: Name or service not known");
+                       },
                        Ok(_) => panic!("Expected error"),
                }
        }
```

Comment (TheBlueMatt) [reply lightning-block-sync/src/http.rs]: Good point. I went with `e.to_string().contains("Name or service not known")` as I'm not 100% sure `to_string` is guaranteed to never change, either.

Comment (TheBlueMatt) [reply lightning-block-sync/src/http.rs]: Turns out its not cross-platform. I think `failed to lookup address information` will be, for now, though.

**Review by valentinewallace — APPROVED** [review]

**Review by jkczyz — APPROVED** [review]

**Pull request merged**